installers/install.sh

```diff
@@ -308,72 +308,21 @@ fetchArtifact () {
   chmod +x $TMPDIR/$TMPEXE
 }
 
-# Use target directory provided by user with no verification or default to
-# one of two commonly used directories. 
-# Ensure they are in PATH and if not use the first writable directory in PATH
-if [ ! -z "$TARGET" ]; then
-  INSTALLDIR=$TARGET
-else
-  if [ -w "/usr/local/bin" ]; then
-    INSTALLDIR="/usr/local/bin"
-  else
-    INSTALLDIR="$HOME/.local/bin"
-  fi
-  # Verify the install directory is in PATH.
-  INPATH=false
-  OLDIFS=$IFS
-  IFS=':'
-  for PATHELEM in $PATH; do 
-    if [ $INSTALLDIR = $PATHELEM ]; then
-      INPATH=true
-      break
-    fi
-  done
-
-  # If the install directory is not in PATH we default to the first
-  # directory in PATH that we have write access to as a last resort.
-  if ! $INPATH; then
-    for PATHELEM in $PATH; do
-      if [ -w $PATHELEM ]; then
-        INSTALLDIR=$PATHELEM
-        break
-      else
-        INSTALLDIR=""
-      fi
-    done
-  fi
-  IFS=$OLDIFS
-fi
-
-if [ -z "$INSTALLDIR" ]; then
-  error "Could not install State Tool to PATH."
-  error "You do not have write access to any directories currently on PATH."
-  error "You can use the '-t' flag to denote an install target, "
-  error "otherwise please ensure you have write permissions to a directory that's on your PATH."
-  exit 1
-fi
-
-# Install to the determined directory.
-info "Installing to $INSTALLDIR"
-if [ ! -e "$INSTALLDIR" ]; then
-  info "NOTE: $INSTALLDIR will be created"
-elif [ -e "$INSTALLDIR/$STATEEXE" ]; then
-  warn "WARNING: overwriting previous installation"
-fi
-if [ ! -z "`which $STATEEXE`" -a "`dirname \`which $STATEEXE\` 2>/dev/null`" != "$INSTALLDIR" ]; then
+if [ ! -z "`which $STATEEXE`" -a "`dirname \`which $STATEEXE\` 2>/dev/null`" != "$CURRENT_INSTALLDIR" ]; then
   warn "WARNING: installing elsewhere from previous installation"
 fi
 userprompt "Continue? [y/N] "
 RESPONSE=$(userinput y)
 case "$RESPONSE" in
   [Yy])
-    # Install.
-    if [ ! -e "$INSTALLDIR" ]; then
-      mkdir -p "$INSTALLDIR" || continue
-    fi
     fetchArtifact
-    info "Installing to $INSTALLDIR..."
-    $TMPDIR/$TMPEXE "$INSTALLDIR"
+    if [ ! -z "$TARGET" ]; then
+      INSTALL_OUTPUT=$($TMPDIR/$TMPEXE "$TARGET")
+    else
+      echo "Installing without target"
+      INSTALL_OUTPUT=$($TMPDIR/$TMPEXE)
```

Comment:
We should probably still print the `INSTALL_OUTPUT`. I also wonder if there is a way to print the installer output when it occurs and still capture it.

This might work: https://stackoverflow.com/a/12451419/4500584

And maybe there is a similar solution for powershell?

Reply:
A modification of your link worked for me in bash. Powershell should be working as well